CommonWallet/Classes/Common/Commands/WalletCommandDecorator.swift

```diff
@@ -0,0 +1,29 @@
+/**
+* Copyright Soramitsu Co., Ltd. All Rights Reserved.
+* SPDX-License-Identifier: GPL-3.0
+*/
+
+import Foundation
+
+public protocol WalletCommandDecoratorProtocol: WalletCommandProtocol {
```

Comment:
How does it suppose to work?

Reply:
The client implements necessary methods from the protocol and new object of the implementation to the wallet builder. Later when any part of the application instantiates a command it is also tries to instantiate a decorator to override the default behavior. Decorator inherits command protocol and also maintains reference for underlying command. The last point is important because it allows to simplify cases like: check something and then execute the original command.